ray-operator/controllers/ray/raycluster_controller.go

```diff
@@ -1257,6 +1272,49 @@ func (r *RayClusterReconciler) calculateStatus(ctx context.Context, instance *ra
 			}
 		}
 
+		suspendStatus := utils.FindRayClusterSuspendStatus(newInstance)
```

Comment:
Can you add comments to explain the logic behind the status transitions here for future reference? Something similar to: https://github.com/ray-project/kuberay/pull/2403#discussion_r1788352417